pkg/test/resourcefixture/contexts/register.go

```diff
@@ -152,8 +147,13 @@ func (rc ResourceContext) getResourceConfig(smLoader *servicemappingloader.Servi
 	panic(fmt.Errorf("no resource config found for kind: %v", rc.ResourceKind))
 }
 
-func (rc ResourceContext) IsAutoGenerated(smLoader *servicemappingloader.ServiceMappingLoader) bool {
-	if rc.IsTFResource {
+func (rc ResourceContext) IsAutoGenerated(smLoader *servicemappingloader.ServiceMappingLoader, u *unstructured.Unstructured) bool {
+	rt, err := testreconciler.ReconcilerTypeForObject(u)
```

Comment:
Similarly, I wondered if we should pass rt in as a parameter.

Or can we make some further refactoring and make reconciler type part of the ResourceContext? As we only call it against the initialUnstruct or the updateUnstruct (i.e. the same kind within the same test case), so I feel it is possible to only get the CRD once per test case. Considering the potential cost of make API calls, I think we should minimize the number of times we call ReconcilerTypeForObject().

Reply:
We could potentially cache the reconciler type in ResourceContext, but that will also reduce flexibility in the future. For example, we may want to add test cases for changing from TF/DCL in create.yaml -> Direct in update.yaml.

Reply:
What I want to avoid is the repetitive CRD file reading, and I assume test cases for changing from TF/DCL in create.yaml -> Direct in update.yaml doesn't require updates in the CRD, but instead an annotation in the object. So can we store the CRD metadata (i.e. information about `TF2CRDLabel` and `DCL2CRDLabel`) into the ResourceContext and retrieve the info from ResourceContext in ReconcilerTypeForObject()?

Reply:
Fair enough. I have proposed a solution to use the GVK map here: https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2837#discussion_r1792643849